web/modules/custom/dpl_fbi/src/Plugin/Field/FieldWidget/CqlSearchWidget.php

```diff
@@ -19,20 +33,398 @@
+      '#open' => FALSE,
+      'video' => [
+        '#type' => 'markup',
+        '#markup' => Markup::create("<video controls width=\"100%\" ><source src=\"/$videoPath\" type=\"video/webm\" /></video>"),
+      ],
+    ];
+
+    $element['filters'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Search filters', [], ['context' => 'DPL material search']),
+      '#attributes' => [
+        'class' => ['material-search-filters'],
+      ],
+
+      'cql' => [
+        '#type' => 'textarea',
+        '#title' => $fieldStorageDefinition->getPropertyDefinition('value')?->getLabel(),
+        '#required' => TRUE,
+        '#default_value' => $items[$delta]->value ?? '',
+        '#weight' => 0,
+      ],
+
+    ];
+
+    if ($this->getSetting('advanced')) {
+      $element['filters'] += [
+        'sort' => [
+          '#type' => 'select',
+          '#weight' => -1,
+          '#title' => $fieldStorageDefinition->getPropertyDefinition('sort')?->getLabel(),
+          '#default_value' => $items[$delta]->sort ?? 'sort.latestpublicationdate.desc',
+          '#options' => [
+            'sort.latestpublicationdate.desc' => $this->t(
+              'By publication date (descending)', [], ['context' => 'DPL material search']
+            ),
+            'sort.latestpublicationdate.asc' => $this->t(
+              'By publication date (ascending)', [], ['context' => 'DPL material search']
+            ),
+            'sort.title.desc' => $this->t(
+              'By title (descending)', [], ['context' => 'DPL material search']
+            ),
+            'sort.title.asc' => $this->t(
+              'By title (ascending)', [], ['context' => 'DPL material search']
+            ),
+            'sort.creator.desc' => $this->t(
+              'By creator (descending)', [], ['context' => 'DPL material search']
+            ),
+            'sort.creator.asc' => $this->t(
+              'By creator (ascending)', [], ['context' => 'DPL material search']
+            ),
+            'relevance' => $this->t(
+              'By relevance', [], ['context' => 'DPL material search']
+            ),
+          ],
+        ],
+        'location' => [
+          '#type' => 'textarea',
+          '#title' => $fieldStorageDefinition->getPropertyDefinition('location')?->getLabel(),
+          '#default_value' => $items[$delta]->location ?? '',
+        ],
+        'sublocation' => [
+          '#type' => 'textarea',
+          '#title' => $fieldStorageDefinition->getPropertyDefinition('sublocation')?->getLabel(),
+          '#default_value' => $items[$delta]->sublocation ?? '',
+        ],
+        'onshelf' => [
+          '#type' => 'checkbox',
+          '#title' => $fieldStorageDefinition->getPropertyDefinition('onshelf')?->getLabel(),
+          '#default_value' => $items[$delta]->onshelf ?? '',
+        ],
+      ];
+    }
+
     return $element;
   }
 
+  /**
+   * AJAX callback, triggered when inputting URL in link field.
```

Comment:
You opted for using AJAX API here. I wonder if the same thing couple be achieved using JavaScript and it that was something you considered.